spec/requests/api/v2/providers/courses_spec.rb

```diff
@@ -151,6 +151,101 @@
     end
   end
 
+  describe 'POST publish' do
+    let(:api_status)   { 200 }
+    let(:api_response) { '{ "result": true }' }
+    let!(:stubbed_manage_courses_api_request) do
+      stub_request(:post, %r{#{Settings.manage_api.base_url}/api/Publish/internal/course/})
+        .to_return(
+          status: api_status,
+          body: api_response
+        )
+    end
+    let(:course) {
+      create(:course,
+             provider: provider,
+             with_site_statuses: [:new],
+             with_enrichments: [:initial_draft])
+    }
+    let(:publish_path) do
+      "/api/v2/providers/#{provider.provider_code}" +
+        "/courses/#{course.course_code}/publish"
+    end
+
+    subject do
+      post publish_path, headers: { 'HTTP_AUTHORIZATION' => credentials }
+      response
+    end
+
+    context 'when unauthenticated' do
+      let(:payload) { { email: 'foo@bar' } }
+
+      it { should have_http_status(:unauthorized) }
+    end
+
+    context 'when user has not accepted terms' do
+      let(:user)         { create(:user, accept_terms_date_utc: nil) }
+      let(:organisation) { create(:organisation, users: [user]) }
+
+      it { should have_http_status(:forbidden) }
+    end
+
+    context 'when unauthorised' do
+      let(:unauthorised_user) { create(:user) }
+      let(:payload)           { { email: unauthorised_user.email } }
+
+      it "raises an error" do
+        expect { subject }.to raise_error Pundit::NotAuthorizedError
+      end
+    end
+
+    context 'when course and provider is not related' do
+      let(:course) { create(:course) }
+
+      it 'raises an error' do
+        expect { subject }.to raise_error ActiveRecord::RecordNotFound
+      end
+    end
+
+    context 'unpublished course with draft enrichment' do
+      let!(:course) {
+        create(:course,
+               provider: provider,
+               with_site_statuses: [:new],
+               with_enrichments: [:initial_draft],
+               age: 17.days.ago)
+      }
+      it 'publishes a course' do
+        # run the POST
+        response = subject
```

Comment:
Ya, I agree that this is weird. I'm not sure how to suggest sorting it. :-/ Actually, I'm just a bit PRed out.

Reply:
Doesn't `response` get magically set when doing a request in request specs?

Reply:
there's some oddness where this assignment causes the POST to actually be run, bit of a hack that needs revisiting. We'll have another look as it isn't right how it is. We'll tackle this after breaking the files up.

Reply:
done 👍